### PR TITLE
fix: [MEW-2257] drop EIP-6963 announceProvider null-detail sentry noise

### PR DESCRIPTION
**Pull request by renerpdev**
## What was wrong

On the token/portfolio pages, a browser wallet extension could crash MEW's wallet-provider discovery. When a buggy or hostile extension announces itself via an `eip6963:announceProvider` event with a **null / malformed payload**, three independent listeners tried to read `.info` off it and threw `TypeError: Cannot read properties of null (reading 'info')` (and a "Cannot destructure property 'info'" variant). These were uncaught and reported to Sentry as if they were app bugs, even though 0 users were actually affected in a way MEW can act on.

Three grouped Sentry issues, one root cause:
- `APP-MEW-WEB-1JM` — bundled `mipd` provider-discovery store
- `APP-MEW-WEB-1JN` — wagmi connector enumeration (`getProviders()`)
- `APP-MEW-WEB-1JG` — MEW's own `providerStore.addProvider`

## What changed

Following the repo's established Sentry noise-suppression pattern, this adds **one narrow `beforeSend` predicate** (`isEip6963NullProviderError` in `src/sentry/extensionNoise.ts`, wired into `src/main.ts`). It drops an error only when BOTH:
1. the message is the null-`info` deref (either shape), AND
2. the stack points at an EIP-6963 provider-discovery frame (`requestProviders` / `createStore` / `getProviders` / `announceProvider` / `addProvider`).

This suppresses all three issues without swallowing unrelated null-`info` dereferences elsewhere in the app. It fails open (never suppresses on the message alone when no stack is present). Backed by 10 new unit tests covering each of the three crash shapes and the negative cases.

**Assumptions / scope notes (decided autonomously, no user confirmation available):**
- The real trigger is untrusted third-party extension input. The `mipd`/`wagmi` crash paths (1JM/1JN) live inside bundled dependencies and cannot be guarded at the source without a dependency bump, which is out of scope for a noise ticket. There is no first-party guard point that stops `mipd` from receiving the event, so suppression at the Sentry boundary is the correct, in-scope fix — consistent with the many prior `extensionNoise.ts` predicates.
- The predicate matches on the raw error `message` + `stack` (like `isExtensionOrProviderError` / `isMetaMaskSdkDecryptError`), since these identifiers survive minification in the production bundle.

## How to test

This is a Sentry-ingestion-boundary change with no user-visible UI difference — the primary verification is the unit suite.

1. `pnpm vitest run tests/unit/sentry/extensionNoise.spec.ts` → all pass (97 tests, 10 new).
2. `pnpm vue-tsc --noEmit -p tsconfig.app.json` → clean.
3. Smoke: load the app (e.g. `/portfolio/token/ethereum`) with a normal browser wallet extension installed; connect flow and wallet list behave exactly as before. Genuine app errors still report to Sentry — only the null-`info` EIP-6963 announce crash is dropped.

## Sentry

- https://myetherwallet-inc.sentry.io/issues/APP-MEW-WEB-1JM
- https://myetherwallet-inc.sentry.io/issues/APP-MEW-WEB-1JN
- https://myetherwallet-inc.sentry.io/issues/APP-MEW-WEB-1JG

## Summary by CodeRabbit

* **Bug Fixes**
  * Reduced unnecessary error reporting for malformed EIP-6963 wallet provider discovery events with missing or invalid provider details.
  * Preserved reporting for unrelated errors, invalid event data, and errors without the expected provider-discovery context.

* **Tests**
  * Added coverage to verify accurate filtering of recognized provider-discovery errors while retaining unrelated error reports.

### CODE REVIEW

**Review by gamalielhere — APPROVED** [review]

File: tests/unit/sentry/extensionNoise.spec.ts [hunk 1]
```diff
@@ -865,15 +871,120 @@ describe('isWalletConnectSubscribeInterruptedError', () => {
...
+      }),
+    ).toBe(false)
+  })
+
+  it('is false for non-matching inputs', () => {
+    expect(isEip6963NullProviderError(null)).toBe(false)
+    expect(isEip6963NullProviderError(undefined)).toBe(false)
+    expect(isEip6963NullProviderError({})).toBe(false)
+    expect(isEip6963NullProviderError('boom')).toBe(false)
+    expect(isEip6963NullProviderError({ message: 42 })).toBe(false)
```
[the remark this answers is not in the document]
Comment (gamalielhere) [reply tests/unit/sentry/extensionNoise.spec.ts:1004]: @renerpdev

File: src/sentry/extensionNoise.ts [hunk 2]
```diff
@@ -381,10 +380,53 @@ export function isBenignPurchaseInfoForbidden(event: unknown): boolean {
...
+// `he.info` read (`Cannot read properties of null (reading 'info')`) and the
+// `({ info }) =>` destructure (`Cannot destructure property 'info' of ...`).
+const NULL_INFO_MESSAGE =
+  /cannot read properties of null \(reading 'info'\)|cannot destructure property 'info' of/i
+// An EIP-6963 provider-discovery frame — retained (non-mangled) in the minified
+// bundle: the bundled `mipd` store (`requestProviders`, `createStore`), wagmi's
+// connector enumeration (`getProviders`), the `eip6963:announceProvider`
+// listeners, and MEW's own `providerStore.addProvider`.
+const EIP6963_DISCOVERY_FRAME =
+  /requestProviders|createStore|getProviders|announceProvider|eip6963|addProvider/i
```
[the remark this answers is not in the document]
Comment (renerpdev) [reply src/sentry/extensionNoise.ts]: Done in 8463bd3. You're right that `createStore` was too broad: it's a zustand internal that wagmi and other stores use well outside provider discovery, so a null-`info` deref passing through one would get suppressed. I dropped the bare `createStore` and anchored the wagmi enumeration crash (1JN) on `createConfig` instead, which is specific to wagmi's provider setup and is present in that stack. The mipd path (1JM) and the first-party `addProvider` path (1JG) still match on their own specific frames, so all three issues stay covered. I went with `createConfig` rather than a strictly EIP-6963-only frame because the 1JN stack has no EIP-6963-named frame at all. Also added the negative test you asked for: a null-`info` error whose stack only passes through a generic `createStore` / `createStoreImpl` is no longer dropped.

File: tests/unit/sentry/extensionNoise.spec.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (renerpdev) [reply tests/unit/sentry/extensionNoise.spec.ts:1004]: This one is a false positive. The test file parses and runs fine: `pnpm vitest run tests/unit/sentry/extensionNoise.spec.ts` is green (98 passing after the latest commit, 97 before it). The final `it` and the surrounding `describe` are properly closed at the end of the file; CodeRabbit flagged this off a truncated diff view and miscounted the braces. @gamalielhere nothing to fix here, the suite has been green the whole time.
